File: solutions/2418. Sort the People/2418.cpp
// Time complexity: O(sort)
// Space complexity: O(n)
class Solution {
 public:
  vector<string> sortPeople(
    const vector<string>& names, 
    const vector<int>& heights
  ) {
    vector<string> ans;
    vector<pair<int, string>> heightAndNames;

    for (int i = 0; i < names.size(); ++i)
      heightAndNames.emplace_back(heights[i], names[i]);

    ranges::sort(heightAndNames, greater<>());

    for (const auto& [_, name] : heightAndNames)
      ans.push_back(name);

    return ans;
  }
};

File: solutions/2418. Sort the People/2418.py
# Time complexity: O(sort)
# Space complexity: O(n)
class Solution:
  def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
    return [
      height for _, height in sorted(
        [(height, name) for name, height in zip(names, heights)],
        reverse=True
      )
    ]